<element name="ray" required="0">
  <element name="scan" required="1">
    <element name="horizontal" required="1">
      <attribute name="samples" type="unsigned int" default="1" required="1"/>
      <attribute name="resolution" type="double" default="1" required="0"/>
      <attribute name="min_angle" type="double" default="0" required="1"/>
      <attribute name="max_angle" type="double" default="0" required="1"/>
    </element> <!-- End Horizontal -->
    <element name="vertical" required="0">
      <attribute name="samples" type="unsigned int" default="1" required="1"/>
      <attribute name="resolution" type="double" default="1" required="0"/>
      <attribute name="min_angle" type="double" default="0" required="1"/>
      <attribute name="max_angle" type="double" default="0" required="1"/>
    </element> <!-- End Vertical -->
  </element> <!-- End Scan -->
  <element name="range" required="1">
    <attribute name="min" type="double" default="0" required="1"/>
    <attribute name="max" type="double" default="0" required="1"/>
    <attribute name="resolution" type="double" default="0" required="0"/>
  </element> <!-- End Range -->
</element> <!-- End Ray -->
